ipa-core/src/cli/playbook/mod.rs

```diff
@@ -106,18 +113,41 @@ pub fn validate_dp(expected: Vec<u32>, actual: Vec<u32>, epsilon: f64, per_user_
 
         let next_expected_f64: f64 = next_expected.unwrap().into();
         let actual_expect_f64: f64 = next_actual.unwrap().into();
+
         let noise_params = NoiseParams {
             epsilon,
+            per_user_credit_cap,
             ell_1_sensitivity: per_user_credit_cap.into(),
             ell_2_sensitivity: per_user_credit_cap.into(),
             ell_infty_sensitivity: per_user_credit_cap.into(),
             dimensions: 256.0, // matches the hard coded number of breakdown keys in oprf_ipa.rs/execute
             ..Default::default()
         };
-
-        let (mean, std) = crate::protocol::dp::noise_mean_std(&noise_params);
-        let same = actual_expect_f64 - mean > next_expected_f64 - 10.0 * std
-            && actual_expect_f64 - mean < next_expected_f64 + 10.0 * std;
+        let same = match dp_mechanism {
+            DpMechanism::Binomial { epsilon: _ } => {
+                let (mean, std) = crate::protocol::dp::binomial_noise_mean_std(&noise_params);
+                actual_expect_f64 - mean > next_expected_f64 - 10.0 * std
+                    && actual_expect_f64 - mean < next_expected_f64 + 10.0 * std
+            }
+            DpMechanism::DiscreteLaplace { epsilon: _ } => {
+                let truncated_discrete_laplace = OPRFPaddingDp::new(
+                    noise_params.epsilon,
+                    noise_params.delta,
+                    noise_params.per_user_credit_cap,
+                )
+                .unwrap();
+
+                let (mean, std) = truncated_discrete_laplace.mean_and_std();
+                let tolerance_factor = 12.0;
```

Comment:
should this be an argument with a default of 12? can we also add a comment explaining what this is and why it's set to 12?

Reply:
sure, will add some comment about why we set it so large. What we want is for this check to fail randomly with very small probability.